### PR TITLE
FreeRTOS TCP

### PR DESCRIPTION
**Pull request by mikewolfram**
Uses new URL for cloning.
Runs a submodule update to get the kernel sources.
Copies the TCP files.

### CODE REVIEW

**Review by salkinium — APPROVED** [review]
Great!

**Comment by mikewolfram** [comment]
Could you trigger a manual run to get the TCP sources?

**Comment by salkinium** [comment]
I already did, and it didn't work, I'm investigating why.

**Comment by salkinium** [comment]
I suspect it didn't commit new files, but that did work before…

**Comment by salkinium** [comment]
This is an issue with the case-insensitive file system on macOS and the case-sensitive FS on Ubuntu. I can reproduce the issue.

File: update.py [hunk 1]
```diff
@@ -18,14 +18,26 @@
     "FreeRTOS/License",
 ]
 
+source_paths_tcp = [
+	"FreeRTOS-Plus/Source/FreeRTOS-PLUS-TCP/portable/BufferManagement",
+	"FreeRTOS-Plus/Source/FreeRTOS-PLUS-TCP/portable/Compiler/GCC",
+	"FreeRTOS-Plus/Source/FreeRTOS-PLUS-TCP/include/*.h",
+	"FreeRTOS-Plus/Source/FreeRTOS-PLUS-TCP/tools",
+	"FreeRTOS-Plus/Source/FreeRTOS-PLUS-TCP/*.c",
+	"FreeRTOS-Plus/Source/FreeRTOS-PLUS-TCP/LICENSE_INFORMATION.txt",
```
Comment (salkinium) [inline update.py:27]: `FreeRTOS-Plus-TCP` was the issue, but due to the case-insensitive macOS file system it works locally, but not in Travis CI.

**Comment by salkinium** [comment]
Fixed.

**Pull request merged**